src/ngraph/runtime/cpu/pass/cpu_assignment.cpp

```diff
@@ -76,6 +77,31 @@ namespace ngraph
                     }
                 }
 
+                template <>
+                void CPUAssignment::ASSIGN_DECL(ngraph::op::Concat)
+                {
+                    auto concat = static_cast<op::Concat*>(node);
+                    size_t num_inputs = node->get_input_size();
+                    bool flag = true;
+
+                    for (size_t i = 0; i < num_inputs; i++)
+                    {
+                        if (node->get_input_element_type(i) == element::f32 &&
+                            (node->get_input_shape(i)).size() == 4)
+                            continue;
+                        else
+                            flag = false;
+                    }
```

Comment:
I think, we don't need to have this `flag` here. ngraph is strongly typed. so it ensures all the arguments have same rank during compilation time. so you can simplify here. Also I think you can allow it as mkldnn op,even if the rank size is 2.

Reply:
Agree. We might need to special case later for performance reasons but API seems broad enough for now.

Reply:
ok so you guys are saying API should be good enough to handle any dimensioned input and also we dont need to check for the element type of the inputs?

Reply:
we will still need element type check. @pruthviIntel was referring to this: https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/op/concat.cpp#L49-L57

You only need to check element_type and rank for the first input tensor. Rest will be the same.

Reply:
@jbobba ok makes sense now. Also, should be just have the input limited to rank 4 only for mkldnn? or as @pruthviIntel mentioned above rank 2 should also be ok

Reply:
I'd suggest doing a quick timing experiment to make sure MKLDNN still performs well on smaller tensors